### PR TITLE
Separate init / ObjectPool.init()

### PR DESCRIPTION
**Pull request by pro100kryto**
ObjectPool.init() allow to initialize partitions after pool creation. It can be used, for example, if you want to have some ObjectFactory witch contains ref of the pool. Like this:
```java
class XObjectFactory implements ObjectFactory<XObject> {
 @Setter
 private ObjectPool<XObject> pool;
 @Override public XObject create() {
   return new XObject(pool); // object requeres pool reference
  }
 ...
}

XObjectFactory factory = new XObjectFactory();
ObjectPool<XObject> pool = new ObjectPool(conf, factory, false); // create pool, but dont init
factory.setPool(pool); // set pool for factory
pool.init(); // fill pool with new objects
```

### CODE REVIEW

File: src/main/java/cn/danielw/fop/ObjectPool.java [hunk 1]
```diff
@@ -11,25 +11,48 @@
  */
 public class ObjectPool<T> {
 
-    private static final Logger logger = Logger.getLogger(ObjectPool.class.getCanonicalName());
+    protected static final Logger logger = Logger.getLogger(ObjectPool.class.getCanonicalName());
```
Comment (pro100kryto) [inline src/main/java/cn/danielw/fop/ObjectPool.java:14]: I think this line is bad, because logger should be private.

**Comment by DanielYWoo** [comment]
You should not have an XFactory to reference the pool. This causes circular dependency between the Pool and the ObjectFactory. The ObjectPool contains the Factory, but the Factory should never be aware of the ObjectPool's existence at all. If you look at other object pool implementation, you will see very similar designs.

**Comment by pro100kryto** [comment]
```java
@AllArgsConstructor
public class Processor {
 private final ObjectPool<Processor> pool;
 ...

 public void process() {
  ...
  if (...) {
   Poolable<Processor> subProcessor = pool.borrowObject();
   subProcessor.getObject().configure(...);
   processAsync( subProcessor );
  }
  ...
 }

 private void processAsync(Poolable<Processor> subProcessor) {
  new Thread(() -> {
   try {
    subProcessor.getObject().process();
   } finally {
    subProcessor.returnObject();
   }
  }).start();
 }

 public void configure(...) { ... }
}
```

**Comment by DanielYWoo** [comment]
In your example, this is exactly a wrong design. The processor depends on the pool, the pool contains the processor. You should have a ProcessorExecutor, which contains a Pool of Processors. Then use ProcessorExecutor.process() or processAsync() to borrow a processor from the pool and execute it.

**Comment by DanielYWoo** [comment]
Close as an invalid requirement.
BTW, the AllArgsConstructor annotation is not recommended, try the builder annotation, or write a constructor. The implicit argument order will harm you by using AllArgsConstructor.

**Pull request closed without merging**